src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -4033,11 +4033,90 @@ uint64_t BlockchainLMDB::get_database_size() const
   return size;
 }
 
-void BlockchainLMDB::fixup()
+void BlockchainLMDB::fixup(fixup_context context)
 {
   LOG_PRINT_L3("BlockchainLMDB::" << __func__);
   // Always call parent as well
-  BlockchainDB::fixup();
+  BlockchainDB::fixup(context);
+
+  if (is_read_only())
+    return;
+
+  if (context.type == fixup_type::calculate_difficulty)
+  {
+    if (context.calculate_difficulty.start_height == 0)
+      context.calculate_difficulty.start_height++;
+
+    uint64_t const start_height = context.calculate_difficulty.start_height;
+
+    std::vector<uint64_t> timestamps;
+    std::vector<difficulty_type> difficulties;
+    {
+      uint64_t offset = start_height - std::min<size_t>(start_height, static_cast<size_t>(DIFFICULTY_BLOCKS_COUNT_V2));
+      if (offset == 0)
+        ++offset;
+
+      if (start_height > offset)
+      {
+        timestamps.reserve  (start_height - offset);
+        difficulties.reserve(start_height - offset);
+      }
+
+      for (; offset < start_height; offset++)
```

Comment:
if start_height is 1 then offset will be 1 but this loop will still execute once.  Should this be `while` instead, with increment at the end of the loop?

Reply:
(1 < 1) executes 0 times.

Reply:
I'm not sure why but I was thinking that `for` loops didn't check the condition going into the first iteration; upon double checking I was mistaken.  Learn something new every day I guess.